sql: TestRaceWithBackfill failed

sql.TestRaceWithBackfill [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10466517?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10466517?buildTab=artifacts#/) on release-22.2 @ [f38ef200f5f1d9f87fba0d5f2956b7fdd7c539b1](https://github.com/cockroachdb/cockroach/commits/f38ef200f5f1d9f87fba0d5f2956b7fdd7c539b1):


```
=== RUN   TestRaceWithBackfill
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestRaceWithBackfill1910564040
    test_log_scope.go:79: use -show-logs to present logs inline
    schema_changer_test.go:337: schema change ALTER TABLE t.test ADD COLUMN x DECIMAL DEFAULT (DECIMAL '1.4') CHECK (x >= 0) took 7.539652455s
    schema_changer_test.go:337: schema change ALTER TABLE t.test DROP pi took 3.96684779s
    schema_changer_test.go:337: schema change CREATE UNIQUE INDEX foo ON t.test (v) took 12.97907424s
    schema_changer_test.go:337: schema change CREATE INDEX bar ON t.test(k) STORING (v) took 9.572502337s
    schema_changer_test.go:417: pq: scrub-index: batch timestamp 1686301951.849521007,0 must be after replica GC threshold 1686301963.443382897,0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestRaceWithBackfill1910564040
--- FAIL: TestRaceWithBackfill (122.29s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRaceWithBackfill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-28655

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -619,6 +619,8 @@ CREATE UNIQUE INDEX vidx ON t.test (v);
 		t.Fatal(err)
 	}
 	tableDesc := desctestutils.TestingGetPublicTableDescriptor(kvDB, keys.SystemSQLCodec, "t", "test")
+	// Disable strict GC TTL enforcement so that we can use AddImmediateGCZoneConfig.
+	defer sqltestutils.DisableGCTTLStrictEnforcement(t, sqlDB)()
 	// Add a zone config for the table so that garbage collection happens rapidly.
 	if _, err := sqltestutils.AddImmediateGCZoneConfig(sqlDB, tableDesc.GetID()); err != nil {
 		t.Fatal(err)

